lib/src/json_schema/validator.dart

```diff
@@ -410,60 +447,78 @@ class Validator {
     if (schema.contains != null) {
       final maxContains = schema.maxContains;
       final minContains = schema.minContains;
-      final containsItems = instance.data.where((item) => Validator(schema.contains).validate(item).isValid).toList();
+
+      var containsItems = [];
+      for (var i = 0; i < instance.data.length; i++) {
+        var item = instance.data[i];
+        var res = _validateAndCaptureEvaluations(schema.contains, Instance(item));
+        if (res) {
+          _setItemAsEvaluated(i);
+          containsItems.add(item);
+        }
+      }
```

Comment:
I could not get a `whereWithIndex` or `mapWithIndex` to work. So back to a good old for-loop.